include/common/DeviceMatrix.h

```diff
@@ -95,6 +95,56 @@ template <typename T> class DeviceMatricesView {
         return input_m_ * input_n_ + output_m_ * output_n_ + (n_matrices_ - 2) * middle_m_ * middle_n_;
     }
 
+    void print(sycl::queue &q, int max_elements_per_matrix = 10) const {
```

Comment:
This should not be here since DeviceMatricesView is a class which lives on the device. Calling print on the device is not possible.

Reply:
moved to io.h as printDeviceMatricesView